frontend/src/js/controllers/featuredChallengeCtrl.js

```diff
@@ -108,6 +111,55 @@
 
         utilities.sendRequest(parameters);
 
+        vm.isStarred = function() {
+            // Get the stars count and user specific starred or unstarred
+            parameters.url = "challenges/" + vm.challengeId + "/";
+            parameters.method = 'GET';
+            parameters.data = {};
+            parameters.token = userKey;
+            parameters.callback = {
+                onSuccess: function(response) {
+                    var details = response.data;
+                    vm.count = details['count'] || 0;
+                    vm.is_starred = details['is_starred'];
+                    if (details['is_starred'] === false) {
+                        vm.caption = 'Star';
+                    } else {
+                        vm.caption = 'Unstar';
+                    }
+                },
+                onError: function() {}
+            };
+            utilities.sendRequest(parameters);
+        };
+        
+        vm.starChallenge = function() {
+            parameters.url = "challenges/" + vm.challengeId + "/";
+            parameters.method = 'POST';
+            parameters.data = {};
+            parameters.token = userKey;
+            parameters.callback = {
+                onSuccess: function(response) {
+                    var details = response.data;
+                    vm.count = details['count'];
+                    vm.is_starred = details['is_starred'];
+                    if (details.is_starred === true) {
+                        vm.caption = 'Unstar';
+                    } else {
+                        vm.caption = 'Star';
+                    }
+                },
+                onError: function(response) {
+                    var error = response.data;
```

Comment:
Is this change what you meant @galipremsagar ?

Reply:
No, that's not what I actually meant. The behaviour should be this:
When you are not logged in and hit **like** in this page, it redirects you to the login page. In our case lets redirect back to the currrent page after login.